### PR TITLE
Export FlutterSemanticsUpdateNotification and improve docs

### PR DESCRIPTION
**Pull request by dnfield**
- Updates the docs on ensureSemantics to better indicate what's synchronous and what isn't
- Adds `FLUTTER_EXPORT` to the notification name so that it doesn't get stripped by the linker

/cc @albertwang0116 @xster

### CODE REVIEW

**Comment by AlbertWang0116** [comment]
LGTM

Thanks for the quick fix :)

File: shell/platform/darwin/ios/framework/Headers/FlutterEngine.h [hunk 1]
```diff
@@ -146,10 +146,16 @@ FLUTTER_EXPORT
  * This method must only be called after launching the engine via
  * `-runWithEntrypoint:` or `-runWithEntryPoint:libraryURI`.
  *
+ * Although this method returns synchronously, it does not guarantee that a
+ * semantics tree is actually available; it only tells the Flutter framework to
+ * build a semantics tree the next it has a good reason to, such as when a frame
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterEngine.h]: Something's missing in this sentence.

File: shell/platform/darwin/ios/framework/Headers/FlutterEngine.h [hunk 2]
```diff
@@ -146,10 +146,16 @@ FLUTTER_EXPORT
  * This method must only be called after launching the engine via
  * `-runWithEntrypoint:` or `-runWithEntryPoint:libraryURI`.
  *
+ * Although this method returns synchronously, it does not guarantee that a
+ * semantics tree is actually available; it only tells the Flutter framework to
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterEngine.h]: a semantics tree is available when this method returns. It synchronously ensures that the next frame the Flutter framework creates will have a semantics tree.

**Comment by xster** [comment]
LGTM

File: shell/platform/darwin/ios/framework/Headers/FlutterEngine.h [hunk 3]
```diff
@@ -146,10 +146,16 @@ FLUTTER_EXPORT
  * This method must only be called after launching the engine via
  * `-runWithEntrypoint:` or `-runWithEntryPoint:libraryURI`.
  *
+ * Although this method returns synchronously, it does not guarantee that a
+ * semantics tree is actually available when the method returns. It
+ * synchronously ensures that the next frame the Flutter framework creates will
+ * have a semantics tree
```
Comment (xster) [inline shell/platform/darwin/ios/framework/Headers/FlutterEngine.h]: period

**Pull request merged**